### DISCUSSION TITLE
Pardon my ignorance but can someone explain #!/bin/bash

### DISCUSSION THREAD

**Discussion by ZacWolf** [unifi-utilities/unifi-common#511] [Q&A]
The various boot.d scripts start with

`#!/bin/bash`

...but when you login to the UDM Pro you don't have access to Bash. It's only available after `unifi-os shell`

How do you manually run a script for testing, etc.?

THANKS!

**Comment by boostchicken** [comment]
if you are still on the 1.x release please don't use the main branch, the scripts in that 1.2.x branch have /bin/sh instead of bash

**Reply by ZacWolf** [reply]
Ah, ok. I upgraded to 2.x, but I haven't done much SSH/CLI work since I did so.  So is bash default in both login and os-shell now?
